lib/spack/spack/version.py

```diff
@@ -304,54 +318,12 @@ def _numeric_lt(self, other):
                     return type(b) == int
                 else:
                     return a < b
+
         # If the common prefix is equal, the one
         # with more segments is bigger.
         return len(self.version) < len(other.version)
 
     @coerced
-    def __lt__(self, other):
-        """Version comparison is designed for consistency with the way RPM
-           does things.  If you need more complicated versions in installed
-           packages, you should override your package's version string to
-           express it more sensibly.
-        """
-        if other is None:
-            return False
-
-        # Coerce if other is not a Version
-        # simple equality test first.
-        if self.version == other.version:
-            return False
-
-        # First priority: anything < develop
-        sdev = self.isdevelop()
-        if sdev:
-            return False    # source = develop, it can't be < anything
-
-        # Now we know !sdev
-        odev = other.isdevelop()
-        if odev:
-            return True    # src < dst
-
-        # now we know neither self nor other isdevelop().
-
-        # Principle: Non-numeric is less than numeric
-        # (so numeric will always be preferred by default)
-        if self.isnumeric():
-            if other.isnumeric():
-                return self._numeric_lt(other)
-            else:    # self = numeric; other = non-numeric
-                # Numeric > Non-numeric (always)
-                return False
-        else:
-            if other.isnumeric():  # self = non-numeric, other = numeric
-                # non-numeric < numeric (always)
-                return True
-            else:  # Both non-numeric
-                # Maybe consider other ways to compare here...
-                return self.string < other.string
```

Comment:
a comment: the whole `if/else` block is not necessary as the string comparison is already happening above (in what used to be`_numeric_lt`) for non-int segments. `spack test versions` pass confirming that this is indeed the case.